demos/noise_suppression_demo/cpp/main.cpp

```diff
@@ -0,0 +1,281 @@
+                                            "devices is shown below). Default value is CPU. "
+                                            "The demo will look for a suitable plugin for device specified.";
+DEFINE_bool(h, false, help_message);
+DEFINE_string(i, "", inp_wav_message);
+DEFINE_string(o, "noise_suppression_demo_out.wav", out_wav_message);
+DEFINE_string(m, "", model_message);
+DEFINE_string(d, "CPU", target_device_message);
+
+static void showUsage() {
+    std::cout << std::endl;
+    std::cout << "noise_suppression_demo [OPTION]" << std::endl;
+    std::cout << "Options:" << std::endl;
+    std::cout << std::endl;
+    std::cout << "    -h           " << help_message << std::endl;
+    std::cout << "    -i INPUT     " << inp_wav_message << std::endl;
+    std::cout << "    -o OUTPUT    " << out_wav_message << std::endl;
+    std::cout << "    -m MODEL     " << model_message << std::endl;
+    std::cout << "    -d DEVICE    " << target_device_message << std::endl;
+}
+
+bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+    // ---------------------------Parsing and validating input arguments--------------------------------------
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        return false;
+    }
+
+    if (FLAGS_i.empty())
+        throw std::logic_error("Parameter -i is not set");
+    if (FLAGS_o.empty())
+        throw std::logic_error("Parameter -o is not set");
+    if (FLAGS_m.empty())
+        throw std::logic_error("Parameter -m is not set");
+
+    return true;
+}
+
+struct RiffWaveHeader {
+    unsigned int riff_tag;       /* "RIFF" string */
+    int riff_length;             /* Total length */
+    unsigned int wave_tag;       /* "WAVE" */
+    unsigned int fmt_tag;        /* "fmt " string (note space after 't') */
+    int fmt_length;              /* Remaining length */
+    short data_format;           /* Data format tag, 1 = PCM */
+    short num_of_channels;       /* Number of channels in file */
+    int sampling_freq;           /* Sampling frequency */
+    int bytes_per_sec;           /* Average bytes/sec */
+    short block_align;           /* Block align */
+    short bits_per_sample;
+    unsigned int data_tag;       /* "data" string */
+    int data_length;             /* Raw data length */
+};
+
+
+const unsigned int fourcc(const char c[4]){
+    return (c[3] << 24) | (c[2] << 16) | (c[1] << 8) | (c[0]);
+}
+void read_wav(const std::string& file_name, std::vector<int16_t>& wave, RiffWaveHeader& wave_header){
+    std::ifstream inp_wave(file_name, std::ios::in|std::ios::binary);
+    if(!inp_wave.is_open())
+        throw std::logic_error("fail to open " + file_name);
+
+    inp_wave.read((char*)&wave_header, sizeof(RiffWaveHeader));
+
+    bool read_ok = true;
+    // make sure it is actually a RIFF file
+
+    if (fourcc("RIFF") != wave_header.riff_tag) {
+        std::cerr << "riff_tag != 'RIFF' for " << file_name << std::endl;
```

Comment:
In case of many bad params (bad samplerate, bad channel num, etc) all bad params will be reported on the first execution at once.
Otherwise you run demo than you fix samplerate and then run demo again and then see that channelnum is also wrong and then fix channel num an so on....

Reply:
Ok. You should create an exception with collected errors then. Let exception handlers decide where to print the error

Reply:
done